crates/fluvio/src/producer/mod.rs

```diff
@@ -297,28 +297,32 @@ impl TopicProducer {
             match topic_spec.get_compression_type() {
                 CompressionAlgorithm::Any => config.compression.unwrap_or_default(),
                 CompressionAlgorithm::Gzip => match config.compression {
-                    Some(Compression::Gzip) | None => Compression::Gzip,
+                    None => Compression::Gzip(GzipLevel::default()),
```

Comment:
I think that here you can use `Default::default()`, which won't require code changes if change the inner elements of this enum variant